packages/webviz-core/src/panels/ImageView/decodings.js

```diff
@@ -114,3 +120,36 @@ export function decodeRGGB(rggb: Uint8Array, width: number, height: number, outp
     }
   }
 }
+
+export function decodeMono8(mono8: Uint8Array, width: number, height: number, output: Uint8ClampedArray) {
+  let inIdx = 0;
+  let outIdx = 0;
+
+  for (let i = 0; i < width * height; i++) {
+    const ch = mono8[inIdx++];
+    output[outIdx++] = ch;
+    output[outIdx++] = ch;
+    output[outIdx++] = ch;
+    output[outIdx++] = 255;
+  }
+}
+
+export function decodeMono16(
+  mono16: Uint8Array,
+  width: number,
+  height: number,
+  is_bigendian: boolean,
+  output: Uint8ClampedArray
+) {
+  let inIdx = 0;
+  let outIdx = 0;
+
+  for (let i = 0; i < width * height; i++) {
+    const val = !is_bigendian ? mono16[inIdx] : mono16[inIdx - 1];
```

Comment:
Did you try `... : mono16[inIdx + 1]` instead? I think `inIdx - 1` is wrong because at the beginning, it will be `mono16[-1]`.